### PR TITLE
mohssen_fathi_solution

### PR DESCRIPTION
**Pull request by mohssenfathi**
Thanks for this month's challenge @blakemerryman . Codables are pretty sweet. Let me know if anything looks off here.

### CODE REVIEW

File: mohssen_fathi_solution/SWAPI.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    catch {
+        print("Error Encoding: \(error.localizedDescription)")
+        return nil
+    }
+}
+
+
+extension DateFormatter {
+    
+    static let iso8601Full: DateFormatter = {
```
Comment (addisonwebb) [inline mohssen_fathi_solution/SWAPI.playground/Contents.swift:126]: I like that you made this an extension on `DateFormatter`. I want to go back and change mine after seeing this!

File: mohssen_fathi_solution/SWAPI.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+
+
+//: A helper function that decodes a `SWCharacter` from raw `Data`.
+//: > Hint: You may need to edit this function. 😉
+func decodeCharacter(from data: Data) -> SWCharacter? {
+    do {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
```
Comment (addisonwebb) [inline mohssen_fathi_solution/SWAPI.playground/Contents.swift:95]: Swift 4.1... nice

**Review by addisonwebb — COMMENTED** [review]
👏 +100 internet points

File: mohssen_fathi_solution/SWAPI.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    let films, species, vehicles, starships: [URL]
+    let created, edited: Date
+    
+    enum Gender: String, Codable {
+        case male
+        case female
+    }
+    
+    /// Alternatively, could make *Color private
+    enum SWColor: String, Codable {
```
Comment (blakemerryman) [inline mohssen_fathi_solution/SWAPI.playground/Contents.swift:51]: This is great. One improvement would be to have separate colors for different things (e.g. `HairColor`, `SkinColor`, etc.)

File: mohssen_fathi_solution/SWAPI.playground/Contents.swift (hunk 2 above)
Comment (blakemerryman) [reply mohssen_fathi_solution/SWAPI.playground/Contents.swift:95]: 🙌

File: mohssen_fathi_solution/SWAPI.playground/Contents.swift (hunk 1 above)
Comment (blakemerryman) [reply mohssen_fathi_solution/SWAPI.playground/Contents.swift:126]: 👌

**Review by blakemerryman — APPROVED** [review]
Nice work! Sorry for the delayed review.

**Comment by blakemerryman** [comment]
Merging this! Feel free to open another PR if you want to make any improvements.

**Pull request merged**